etna/core/models/mixins.py

```diff
@@ -25,3 +30,47 @@ class ContentWarningMixin(models.Model):
 
     class Meta:
         abstract = True
+
+
+class NewLabelMixin(models.Model):
+    """Mixin to allow editors to toggle 'new' label to be applied on-publish"""
+
+    mark_new_on_next_publish = models.BooleanField(
+        verbose_name="mark this page as 'new' when published",
+        default=True,
+    )
+
+    newly_published_at = models.DateField(
+        editable=False,
+        default=None,
+        null=True,
+    )
+
+    new_label_display_for_days = 21
+
+    def save(self, *args, **kwargs):
+        if self.live and self.mark_new_on_next_publish:
+            self.mark_new_on_next_publish = False
+            self.newly_published_at = timezone.now().date()
+        return super().save(*args, **kwargs)
+
+    @cached_property
+    def is_newly_published(self):
+        expiry_date = timezone.now().date() - timedelta(
+            days=self.new_label_display_for_days
+        )
+        if self.newly_published_at is not None:
+            if self.newly_published_at > expiry_date:
+                return True
+        return False
+
+    promote_panels = [
+        MultiFieldPanel(
+            [
+                FieldPanel("mark_new_on_next_publish"),
+            ]
+        ),
```

Comment:
@jamesbiggs Is the `MultiFieldPanel` necessary here? I would normally only expect to see them used to group multiple fields, and a `heading="Some heading"` value to group them?